### PR TITLE
_ModalScope has local copies of route state

### LINKED ISSUE #744: _ModalScope has local copies of route state
_DropDownMenu takes a pointer to its route, and drives everything directly from the route's members.
_ModalScope goes the other way, and takes copies of everything from its route.

We should pick a pattern and go with it. Probably just making _ModalScope have a pointer to the route and drive everything through that instead of having multiple copies of these values.

### PR DESCRIPTION
**Pull request by abarth**
Now we just use the state from the Route.

Fixes #744

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/routes.dart [hunk 1]
```diff
@@ -305,24 +295,17 @@ class _ModalScope extends StatefulComponent {
 class _ModalScopeState extends State<_ModalScope> {
   void initState() {
     super.initState();
-    config.performance?.addStatusListener(_performanceStatusChanged);
-    config.forwardPerformance?.addStatusListener(_performanceStatusChanged);
+    config.route.performance?.addStatusListener(_performanceStatusChanged);
+    config.route.forwardPerformance?.addStatusListener(_performanceStatusChanged);
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/routes.dart]: what about when the forward performance changes?

Comment (Hixie) [inline packages/flutter/lib/src/widgets/routes.dart]: ooh never mind. it's a proxy and handles that transparently.

**Comment by Hixie** [comment]
lgtm

File: packages/flutter/lib/src/widgets/routes.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/routes.dart]: Yeah

**Pull request merged**